File: ValidateUserCredentialsandPerformLogin/app/src/main/java/com/example/validateusercredentialsandperformlogin/MainActivity.java
package com.example.validateusercredentialsandperformlogin;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {
    private Button btnGetStarted;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        btnGetStarted=findViewById(R.id.btnGetStarted);
        btnGetStarted.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent =new Intent(MainActivity.this, LoginScreen.class);
                startActivity(intent);
            }
        });
    }
}

File: RxJavaI/app/src/main/java/dev/kingbond/rxjavai/MainActivity.kt
package dev.kingbond.rxjavai

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
import io.reactivex.rxjava3.core.Observable
import io.reactivex.rxjava3.core.Scheduler
import io.reactivex.rxjava3.schedulers.Schedulers
import kotlinx.android.synthetic.main.activity_main.*
import org.reactivestreams.Subscriber
import java.util.concurrent.TimeUnit

class MainActivity : AppCompatActivity() {

    private val TAG: String = "Hello"
    private var list = ArrayList<StudentClass>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnJust.setOnClickListener {
            Observable.just("Message")
                .subscribe(
                    { value -> Log.d(TAG, "Proceed to my name") },
                    { error -> Log.d(TAG, "Some error occured") },
                    { Log.d(TAG, "My name is Mausam Singh") }
                )
        }

        bntfromArrayButton.setOnClickListener {
            Observable.fromArray("Tokyo", "Ghoul", "Hello", "World")
                .subscribe {
                    Log.d(TAG, it)
                }
        }

        btnRangeButton.setOnClickListener {
            Observable.intervalRange(20L, 20L, 0L, 1L, TimeUnit.SECONDS)
                .subscribe() {
                    Log.d(TAG, "$it")
                }

        }


        btnListOfStudnet.setOnClickListener {
            list.clear()

            for (i in 0..99) {
                if (i >= 0 && i <= 35) list.add(StudentClass(i, "Mausam"))
                else if (i >= 36 && i <= 70) list.add(StudentClass(i, "Tokyo"))
                else if (i >= 71 && i < 80) list.add(StudentClass(i, "Attack on Titan"))
                else list.add(StudentClass(i, "Roberto Juice"))
            }

           var newList=ArrayList<StudentClass>()

//            newList.flatMap(Observable::fromIterable)
//                .filt

        }


    }


}

